File: 9.Palindrome Number/palindromeNumber.go
package main

import (
	"fmt"
	"math"
)

func isPalindrome(x int) bool {
	if x < 0 {
		return false
	}
	length := int(math.Log10(float64(x)))
	temp := x
	for i := 0; i < length+1; i++ {
		pow := int(math.Pow(10, float64(length-i)))
		if x/pow != temp%10 {
			return false
		}
		x -= pow * (x / pow)
		temp /= 10
	}
	return true
}

func main() {
	fmt.Println(isPalindrome(121))
}

File: 9.Palindrome Number/palindromeNumber.py
# def isPalindrome(x: int) -> bool:
#     if x < 0 or (x > 0 and x % 10 == 0):
#         return False

#     revert = 0

#     while x > revert:
#         revert = revert * 10 + x % 10
#         x = x // 10

#     return x == revert or x == revert // 10


def isPalindrome(x: int) -> bool:
    return True if (x > 0 and str(x) == str(x)[::-1]) else False


print(isPalindrome(96969))
